### PR TITLE
feat(core): support ssh build arg in DockerImageAsset

### PR DESCRIPTION
**Pull request by jackwboynton**
Adds support for the docker build --ssh flag for specifying ssh agent socket or ssh keys for ecr DockerImageAsset

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/cloud-assembly-schema/lib/assets/docker-image-asset.ts [hunk 1]
```diff
@@ -63,6 +63,15 @@ export interface DockerImageSource {
...
   readonly dockerBuildArgs?: { [name: string]: string };
 
+  /**
+   * SSH agent socket or keys
+   *
+   * Requires building with docker buildkit.
+   *
+   * @default - No ssh flag is set
+   */
+  readonly dockerBuildSSH?: string;
```
Comment (mrgrain) [inline packages/aws-cdk-lib/cloud-assembly-schema/lib/assets/docker-image-asset.ts:73]:
```suggestion
  readonly dockerBuildSsh?: string;
```

I know the PR needs clean-up anyway, but let's go with this casing everywhere please!

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/cloud-assembly-schema/lib/assets/docker-image-asset.ts (hunk 1 above)
Comment (JackWBoynton) [reply packages/aws-cdk-lib/cloud-assembly-schema/lib/assets/docker-image-asset.ts]: Changed!

File: packages/@aws-cdk/cx-api/FEATURE_FLAGS.md
Comment (mrgrain) [inline packages/@aws-cdk/cx-api/FEATURE_FLAGS.md:1]: Please revert the changes to this file.

**Comment by mrgrain** [comment]
e7327cde-b87f-46d6-a42b-f23b5870cf0e

**Comment by mrgrain** [comment]
@mergifyio update

File: packages/@aws-cdk/cx-api/FEATURE_FLAGS.md
Comment (JackWBoynton) [reply packages/@aws-cdk/cx-api/FEATURE_FLAGS.md:1]: reverted.

**Review by mrgrain — APPROVED** [review]

**Comment by pepito-j** [comment]
Hello. Just wanted to chime in as I noticed this was pushed, but it looks like the --ssh flag still isn't getting propagated in. The results are below but I believe this section still needs to be changed: 
- https://github.com/aws/aws-cdk/blob/27b720135e30d58732997a5a5e172daaa9453381/packages/cdk-assets/lib/private/handlers/container-images.ts#L187-L200

Doesn't seem like we're picking up buildSsh options and instead are still just leaving it empty.


Local Tests:
==

CDK Library Version 2.91.0
CDK CLI Version 2.91.0
NodeJs v16
Environment: Cloud9 / Amazon Linux 2

Test App:
```
const app = new cdk.App();
const stack = new cdk.Stack(app, 'Stack', { env: { region: 'us-east-1'  }})
new cdk.aws_ecr_assets.DockerImageAsset(stack, 'Image', {
  directory: path.join(__dirname, '../', 'images/test/'), # Just a Dockerfile in here
  buildSsh: 'default',
  buildSecrets: { 
    SECRET: cdk.DockerBuildSecret.fromSrc(path.join(path.join(__dirname, '../README.md'),)) 
  }
})
```
Dockerfile
```
# syntax=docker/dockerfile:1
FROM alpine
RUN apk add --no-cache openssh-client git
RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
RUN --mount=type=ssh ssh-add -l > ~/test.txt
```

Partial Debug Output:
```
[21:10:17] 2 total assets, 1 still need to be published
[21:10:17] Retrieved account ID 848135204948 from disk cache
[21:10:17] Waiting for stack CDKToolkit to finish creating or updating...
Stack:  start: Building ca115e1fc5a469db02217d34c4e6c89902a04ad19846806664444b229db52144:current_account-us-east-1
[21:10:17] Stack:  debug: docker login --username AWS --password-stdin https://848135204948.dkr.ecr.us-east-1.amazonaws.com
[21:10:17] Stack:  debug: docker inspect cdkasset-ca115e1fc5a469db02217d34c4e6c89902a04ad19846806664444b229db52144
[21:10:17] Stack:  build: Building Docker image at /home/ec2-user/environment/dockerTest/cdk.out/asset.ca115e1fc5a469db02217d34c4e6c89902a04ad19846806664444b229db52144
[21:10:17] Stack:  debug: docker build --secret id=SECRET,src=/home/ec2-user/environment/dockerTest/README.md --tag cdkasset-ca115e1fc5a469db02217d34c4e6c89902a04ad19846806664444b229db52144 .
........
#10 [stage-0 3/4] RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
#10 sha256:72821f6cf384c419894b47fc9a2f8eb07ca007d93d39da5b3f8912e46257564c
#10 CACHED

#11 [stage-0 4/4] RUN --mount=type=ssh ssh-add -l > ~/test.txt
#11 sha256:e81ab27a7f76dd7e206dc88e36e4c938c860ab988b3e315001516b45f176899b
#11 0.437 Error connecting to agent: No such file or directory
#11 ERROR: executor failed running [/bin/sh -c ssh-add -l > ~/test.txt]: exit code: 2
------
 > [stage-0 4/4] RUN --mount=type=ssh ssh-add -l > ~/test.txt:
------
executor failed running [/bin/sh -c ssh-add -l > ~/test.txt]: exit code: 2

 ❌ Deployment failed: Error: Failed to build asset 5f16471d2d39322f4e45553c2c2bb34e775e7926016766181ad7ce0fedb5f8b7:current_account-us-east-1
    at Deployments.buildSingleAsset (/home/ec2-user/environment/dockerTest/node_modules/aws-cdk/lib/api/deployments.ts:593:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.buildAsset (/home/ec2-user/environment/dockerTest/node_modules/aws-cdk/lib/cdk-toolkit.ts:201:7)
    at /home/ec2-user/environment/dockerTest/node_modules/aws-cdk/lib/util/work-graph.ts:108:11
```

Notice that "docker build --secret id=SECRET,src=/home/ec2-user/environment/dockerTest/README.md --tag cdkasset-ca115e1fc5a469db02217d34c4e6c89902a04ad19846806664444b229db52144 ." doesn't have the "--ssh" flag still.

**Comment by JackWBoynton** [comment]
Thanks for the catch! I have a PR for the fix [here](https://github.com/aws/aws-cdk/pull/26846)
